### PR TITLE
Add ability to filter logs using bytes for topics

### LINKED ISSUE #185: Supply log topic as bytes
### What was wrong?

It's pretty natural to want to supply a log topic as a 32-byte `bytes` value instead of a hex string.

Right now, only hex-encoded strings of 32 bytes are supported.

### How can it be fixed?

Add tests that 32-byte values will be accepted, like these, but with a `decode_hex()`:
https://github.com/ethereum/eth-tester/blob/035dde2c4ebc10a205748b37b940cb519251d9da/tests/core/validation/test_inbound_validation.py#L160-L165

Add tests that 31 or 33-byte values are not accepted.

Add tests that the logs are correctly found when a bytes-value is supplied to `create_log_filter()`: https://github.com/ethereum/eth-tester/blob/2098d83b41141b4102fe9152308529d7c0a67988/eth_tester/utils/backend_testing.py#L1145

Implement any conversions needed to make ^ pass.

### PR DESCRIPTION
**Pull request by reedsa**
### What was wrong?

Closes #185 

### To-Do:

- [x] Add entry to the [release notes](https://github.com/ethereum/eth-tester/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Cute animal picture

### CODE REVIEW

**Review by reedsa — COMMENTED** [review]
@kclowes @fselmo @pacrob @wolovim Based on the issue description, I think these tests should cover what is needed. Before I get into the code I wanted to see if I missed anything thus far or if there are other ideas around this.

**Review by fselmo — APPROVED** [review]
I think this is a great addition. Nothing blocking on my end, just some thoughts.

File: eth_tester/normalization/inbound.py [hunk 1]
```diff
@@ -27,15 +27,21 @@
...
 
 def normalize_topic(topic):
     if topic is None:
         return None
-    else:
+    if is_32byte_hex_string(topic):
         return decode_hex(topic)
+    if is_32byte_string(topic):
+        return topic
+    else:
```
[the remark this answers is not in the document]
Comment (reedsa) [reply eth_tester/normalization/inbound.py:43]: Nice catch, I like your suggestion.

File: eth_tester/validation/inbound.py [hunk 2]
```diff
@@ -47,12 +48,16 @@
 )
 
 
+def is_32byte_string(value):
```
[the remark this answers is not in the document]
Comment (reedsa) [reply eth_tester/validation/inbound.py:51]: Agreed, `is_32_bytes()` is best here. The function name below is slightly bothersome but I think I'll leave that as is.

File: eth_tester/validation/inbound.py [hunk 3]
```diff
@@ -156,7 +161,8 @@ def validate_filter_params(from_block, to_block, address, topics):
 
     invalid_topics_message = (
         "Topics must be one of `None`, an array of 32 byte hexadecimal encoded "
-        "strings, or an array of arrays of 32 byte hexadecimal strings"
+        "strings, an array of arrays of 32 byte hexadecimal strings, or an array of 32 "
+        "byte strings"
```
Comment (fselmo) [inline eth_tester/validation/inbound.py:165]: I missed the "byte strings" here too. This should maybe read more like 

```py
invalid_topics_message = (
    "Topics must be one of `None`, an array of bytes with length 32 (or their hexadecimal "
    "string equivalents), or an array of arrays of bytes with length 32 (or their hexadecimal "
    "string equivalents)."
)
```

Best I could come up with 😅 but something along those lines?

**Review by pacrob — APPROVED** [review]
once `invalid_topics_message` is clarified, lgtm!

**Pull request merged**